src/index.ts

```diff
@@ -163,19 +178,71 @@ export function parseVersion(client: InstallableClient, rawVersionRange: string)
     return new semver.Range(rawVersionRange);
 }
 
-function getActionInputs(): ClientsToInstall {
+function getActionInputs(): SourceAndClients {
     const clientsToInstall: ClientsToInstall = {};
+    let source = "mirror";
+    let pat = "";
 
-    for (const client of Object.values(Inputs)) {
-        const clientVersion = ghCore.getInput(client);
+    for (const input of Object.values(Inputs)) {
+        if (input === Inputs.SOURCE) {
+            source = ghCore.getInput(input);
+            if (source !== "mirror" && source !== "github") {
+                throw new Error(`❌ "${source}" is not a valid input. Valid inputs are "mirror" or "github"`);
+            }
+        }
+        else if (input === Inputs.GITHUB_PAT && source === "github") {
+            pat = ghCore.getInput(input);
+            if (!pat) {
+                throw new Error(`❌ Input "${input}" must be provided to install the tools from Github.`);
+            }
+        }
+        else if (input !== Inputs.GITHUB_PAT) {
+            const clientVersion = ghCore.getInput(input);
+            if (clientVersion) {
+                ghCore.info(`ℹ️ Provided source is "${source}". Tools will be installed from "${source}".`);
+                ghCore.info(`Installing ${input} matching version "${clientVersion}"`);
+                clientsToInstall[input] = parseVersion(input, clientVersion);
+            }
+        }
+    }
+
+    return {
+        source, clientsToInstall,
+    };
+}
 
-        if (clientVersion) {
-            ghCore.info(`Installing ${client} matching version "${clientVersion}"`);
-            clientsToInstall[client] = parseVersion(client, clientVersion);
+/**
+ * Check if the provided client is supported in the provided source
+ *
+ * @param source Provided source to install clients from
+ * @param clientsToInstall List of clients the need to be installed
+ */
+function checkIfProvidedClientSupported(source: string, clientsToInstall: ClientsToInstall): void {
+    const githubUnSupportedClient = [];
+    const mirrorUnSupportedClient = [];
+    for (const [ client_ ] of Object.entries(clientsToInstall)) {
+        const client = client_ as InstallableClient;
+        if (source === "github" && !ClientDetailOverrides[client]?.githubRepositoryPath) {
```

Comment:
"github" and "mirror" strings should really be in a constant/enum somewhere.

Reply:
Okay :+1: